heartbeat/monitors/active/http/respbody.go

```diff
@@ -129,8 +125,9 @@ func readPrefixAndHash(body io.ReadCloser, maxPrefixSize int) (respSize int, pre
 	}
 
 	// We discard the body if it is not valid UTF-8
-	if utf8.Valid(prefixBuf[:prefixWriteOffset]) {
-		prefix = string(prefixBuf[:prefixWriteOffset])
+	prefixBytes := prefixBuf.Bytes()
+	if utf8.Valid(prefixBytes) {
+		prefix = string(prefixBytes)
 	}
```

Comment:
Wouldn't it make more sense to validate utf in for loop , chunk by chunk? and even if one chunk is , break the loop and discard it. That way, it won't have to read remaining body.

Reply:
I'm not sure if that would work with multi-byte characters. In fact, I think the current code is probably wrong as well where we hit a multi-byte boundary on the 100th megabyte.

Looking deeper into this, I don't see a valid reason to discard non-UTF-8 input in the first place, at least not at this point. The body matching feature of heartbeat works fine with non-UTF-8.

That being said, the JSON encoding process seems to omit non-UTF8 strings in events, so a non-UTF-8 body is silently deleted before being output anywhere (including Elasticsearch). This is fine as ES requires UTF-8 anyway. 

I've opened https://github.com/elastic/beats/issues/15647 since this is a separate issue.

I've also removed the UTF-8 detection from this code for now since it was serving no functional purpose.

Reply:
I'm pretty sure that would break on multi-byte char boundaries. Thinking about it, even the current code is probably broken on multi-byte unicode > 100MiB where the 100MiB limit falls in the middle of a char.

Looking into it, our unicode detection does nothing here, I've removed it in this PR to keep things simple. I've opened https://github.com/elastic/beats/issues/15647 to follow-up on this.